src/module/actor/lancer-actor.ts

```diff
@@ -872,26 +891,66 @@ export class LancerActor extends Actor {
         dr.overshield.max = mm.MaxHP; // as good a number as any I guess
 
         // Depending on type, setup derived fields more precisely as able
-        if (mm.Type != EntryType.PILOT) {
-          let robot = mm as Mech | Npc | Deployable;
-
+        if (!is_reg_pilot(mm)) {
           // All "wow, cool robot" type units have these
-          dr.save_target = robot.SaveTarget;
-          dr.heat.max = robot.HeatCapacity;
-          dr.heat.value = robot.CurrentHeat;
+          dr.save_target = mm.SaveTarget;
+          dr.heat.max = mm.HeatCapacity;
+          dr.heat.value = mm.CurrentHeat;
+
+          // If the Size of the ent has changed since the last update, set the
+          // protype token size to the new size
+          const cached_token_size = this.data.token.flags[game.system.id]?.mm_size;
+          if (!cached_token_size || cached_token_size !== mm.Size) {
+            const size = mm.Size <= 1 ? 1 : mm.Size;
+            this.data.token.update({
+              width: size,
+              height: size,
+              flags: {
+                "hex-size-support": {
+                  borderSize: size,
+                  altSnapping: true,
+                  evenSnap: !(size % 2),
+                },
+                [game.system.id]: {
+                  mm_size: size,
+                }
+              },
+            });
+          }
+
+          // And copying Bolts' work, do the same thing for the token image 
+          // with Retrograde's work
+          const cached_frame = this.data.token.flags[game.system.id]?.mm_size;
```

Comment:
tsserver tells me this `cached_frame` is never used?

Reply:
Find references only shows this location for me.

Reply:
Also, the definitions for Token flags are at the bottom of token.ts.

Reply:
Oops, yeah... left part of an old approach in.